### DISCUSSION TITLE
Track changes of exported data of packages

### DISCUSSION THREAD

**Discussion by psychelzh** [ropensci/targets#352] [Help] [answered]
Currently, `tar_option_set(imports = "pkg")` does well in tracking changes of functions in a packages, but I do not find it is the same for the exported data of a package. I think there can be some cases when the changes in exported data should be tracked.

**Comment by wlandau** [comment, marked as answer]
Unfortunately, the environment returned by `getNamespace()` does not include datasets, which is how `targets` would handle this. In cases where you do need to track a dataset from a package, you can assign it to a global variable in `_targets.R`.

``` r
library(targets)
 
tar_script({
library(targets)
library(gapminder)
dataset <- gapminder
list(tar_target(data_target, dataset))
})
  
tar_visnetwork()
```

![](https://i.imgur.com/kEGuXk0.png)

<sup>Created on 2021-03-21 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

**Reply by psychelzh** [reply]
Is it okay to call `data(gapminder, package = "gapminder")` instead? In this way, one can use the original name `gapminder` to track it.

**Reply by wlandau** [reply]
You can certainly try it and see if you get the right dependency graph.

**Reply by psychelzh** [reply]
Okay, have confirmed it also works! Thank you very much!
